cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java

```diff
@@ -239,4 +239,25 @@ public void replicatorBuilderAddsCookieInterceptorDefaultPort() throws Exception
        assertCookieInterceptorPresent(p, "name=%F0%9F%8D%B6&password=%F0%9F%8D%B6");
     }
 
+    @Test
+    public void replicatorBuilderAddsCookieInterceptorSpecialCreds() throws Exception {
```

Comment:
Not sure if the name is that meaningful how about `replicatorBuilderAddsCookieInterceptorCredsPercentEncoded`?
And a comment to say that the test is to check that double-encoding isn't occurring.

The existing tests above aren't very well named (replicatorBuilderAddsCookieInterceptorCustomPort etc) but this is outside of the PR scope